File: business/business-oj/src/main/java/com/glowxq/oj/judge/enums/JudgeMode.java
package com.glowxq.oj.judge.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum JudgeMode implements BaseType {
    TEST("test", "测试模式"),
    DEFAULT("default", "默认模式"),
    SPJ("spj", "SPJ模式"),
    INTERACTIVE("interactive", "交互式模式");

    private final String mode;

    private final String name;



    public static JudgeMode getJudgeMode(String mode) {
        for (JudgeMode judgeMode : JudgeMode.values()) {
            if (judgeMode.getMode().equals(mode)) {
                return judgeMode;
            }
        }
        return null;
    }

    @Override
    public Object getCode() {
        return mode;
    }
}

File: oj-web/src/enums/oj/judge/JudgeMode.ts
import type { BaseEnum } from '@/enums/base';

// 评测模式枚举接口
interface JudgeModeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

// 实现评测模式枚举对象
export const JudgeMode: { [key: string]: JudgeModeEnum } = {
  TEST: {
    code: 'test',
    name: '测试模式',
    tooltip: '测试代码运行情况',
    text: '仅测试代码运行结果，不进行实际评测'
  },
  DEFAULT: {
    code: 'default',
    name: '默认模式',
    tooltip: '标准输入输出评测',
    text: '使用标准输入输出方式评测提交的代码'
  },
  SPJ: {
    code: 'spj',
    name: 'SPJ模式',
    tooltip: '特殊评测模式',
    text: '使用特判程序评测提交代码的结果'
  },
  INTERACTIVE: {
    code: 'interactive',
    name: '交互式模式',
    tooltip: '交互式评测',
    text: '提交的代码需要与评测程序进行实时交互'
  }
};

// 添加 matchCode 方法
export function matchCode(code: string): JudgeModeEnum | null {
  return Object.values(JudgeMode).find(item => item.code === code) || null;
} 